### PR TITLE
Optimize helm-operator build process

### PR DESCRIPTION
**Pull request by tschuy**
Download fewer unnecessary packages for building helm and tiller, and build helm instead of downloading a precompiled version.

### CODE REVIEW

File: images/helm-operator/Dockerfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ FROM centos:7 as build
 RUN yum -y update && yum clean all
 
 RUN mkdir -p /go && chmod -R 777 /go && \
-    yum -y install epel-release && yum -y install hg git golang glide && yum group install -y "Development Tools"
+    yum -y install epel-release && yum -y install protobuf hg git golang glide make
```
Comment (chancez) [inline images/helm-operator/Dockerfile]: I thought kubectl was one too?

File: images/hive/Dockerfile [hunk 2]
```diff
@@ -14,18 +14,8 @@ RUN set -x \
...
-
-# Install PostgreSQL JDBC
-RUN set -x \
-    && curl -fSLs -o "$HIVE_HOME/lib/$POSTGRESQL_JDBC_JAR" "https://jdbc.postgresql.org/download/$POSTGRESQL_JDBC_JAR"
-
-# Install MySQL JDBC
-RUN set -x \
-    && curl -fSLs "https://dev.mysql.com/get/Downloads/Connector-J/$MYSQL_JDBC_VERSION.tar.gz" | tar -zx --strip-components=1 -C "$HIVE_HOME/lib" "$MYSQL_JDBC_VERSION/$MYSQL_JDBC_JAR"
+# Install PostgreSQL and MySQL JDBCs
+RUN yum install -y mysql-connector-java postgresql-jdbc && yum clean all && rm -rf /var/cache/yum
```
Comment (chancez) [inline images/hive/Dockerfile]: These are definitely not going to necessarily be in the right spots, so you probably need to actually validate this with a mysql/postgresql setup.

Comment (chancez) [reply images/hive/Dockerfile]: Eg: "$HIVE_HOME/lib/$POSTGRESQL_JDBC_JAR"

We probably just need to make sure that where these go ends up in our Java classpath, and all these wrapper scripts correctly use that.

File: images/helm-operator/Dockerfile (hunk 1 above)
Comment (tschuy) [reply images/helm-operator/Dockerfile]: ah, it looks like I wasn't looking at `epel`. There's a 1.7.5 copy of `kubectl` in there.

I don't see it in brew, which is worth noting.

Comment (chancez) [reply images/helm-operator/Dockerfile]: 1.5 is probably too old

**Pull request merged**